doc/man/man3/optional/crypto_hmac_sha512.3monocypher

```diff
@@ -0,0 +1,162 @@
+.Dd December 5, 2019
+.Dt CRYPTO_HMAC_SHA512 3MONOCYPHER
+.Os
+.Sh NAME
+.Nm crypto_hmac_sha512 ,
+.Nm crypto_hmac_sha512_init ,
+.Nm crypto_hmac_sha512_update ,
+.Nm crypto_hmac_sha512_final
+.Nd cryptographic hash-based message authentication code with SHA-512
+.Sh SYNOPSIS
+.In monocypher.h
+.Ft void
+.Fo crypto_hmac_sha512
+.Fa "uint8_t hmac[64]"
+.Fa "const uint8_t *key"
+.Fa "size_t key_size"
+.Fa "const uint8_t *message"
+.Fa "size_t message_size"
+.Fc
+.Ft void
+.Fo crypto_hmac_sha512_init
+.Fa "crypto_hmac_sha512_ctx *ctx"
+.Fa "const uint8_t *key"
+.Fa "size_t key_size"
+.Fc
+.Ft void
+.Fo crypto_hmac_sha512_update
+.Fa "crypto_hmac_sha512_ctx *ctx"
+.Fa "const uint8_t *message"
+.Fa "size_t message_size"
+.Fc
+.Ft void
+.Fo crypto_hmac_sha512_final
+.Fa "crypto_hmac_sha512_ctx *ctx"
+.Fa "uint8_t hmac[64]"
+.Fc
+.Sh DESCRIPTION
+HMAC with SHA-512 is a cryptographically secure message authentication
+code (MAC),
+provided to enable compatibility with other cryptographic systems.
+It is generally recommended to use
+.Xr crypto_blake2b_general 3monocypher
+instead,
+as it performs faster on x86_64 CPUs.
+.Pp
+The arguments are:
+.Bl -tag -width Ds
+.It Fa hmac
+The output MAC,
+which is always 64 bytes long.
+.It Fa key
+Some secret key.
+One cannot predict the final hash without it.
+Users may want to wipe the key with
+.Xr crypto_wipe 3monocypher
+once they are done with it.
+.It Fa key_size
+Length of
+.Fa key ,
+in bytes.
+32 is a good default.
+Keys longer than 128 bytes will be reduced to 64 bytes by hashing
+the key with SHA-512.
+.It Fa message
+The message to compute the HMAC for.
+May overlap with
+.Fa hmac .
+May be
+.Dv NULL
+if
+.Fa message_size
+is 0.
+.It Fa message_size
+Length of
+.Fa message ,
+in bytes.
+.El
+.Pp
+An incremental interface is provided.
+It is useful for handling streams of data or
+large files without using too much memory.
+This interface uses three steps:
+.Bl -bullet
+.It
+initialisation with
+.Fn crypto_hmac_sha512_init ,
+which sets up a context with the hashing parameters;
+.It
+update with
+.Fn crypto_hmac_sha512_update ,
+which hashes the message chunk by chunk, and keep the intermediary
+result in the context;
+.It
+and finalisation with
+.Fn crypto_hmac_sha512_final ,
+which produces the final hash.
+The
+.Ft crypto_hmac_sha512_ctx
+is automatically wiped upon finalisation.
+.El
+.Pp
+.Fn crypto_hmac_sha512
+is a convenience function that
+performs
+.Fn crypto_hmac_sha512_init ,
+.Fn crypto_hmac_sha512_update ,
+and
+.Fn crypto_hmac_sha512_final .
+.Pp
+Use
+.Xr crypto_verify64 3monocypher
+to compare MACs.
```

Comment:
Actually, the main use of HMAC is the same as Poly1305: authenticating messages. For this, a 16 byte hash is more than enough. We can truncate the output to 16 bytes, and use `crypto_verify16(3)` to compare the MACs. It may or may not be worth laying out explicitly.

If you're using it to implement something like Noise however, you probably need the full output.